### PR TITLE
Feature/tkx 64 review nswag and tests

### PR DESCRIPTION
**Pull request by emanuelgaspar**
All aspects of the REST api client generation have been reviewed:

- open api definition for two aggregate methods we have access to (OHLCV and VWAP).
- nswag configuration
- code generation (http client registration no longer requires a T4 template)
- integration tests for reference data
- integration tests for the active market data endpoints: Aggregates OHLCV and VWAP.
- integration test to ensure the GenerateApiClient flag is disabled at origin.

### CODE REVIEW

File: src/Trakx.Kaiko.ApiClient/Registration/HttpClientRegistration.cs [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+using Polly.Contrib.WaitAndRetry;
+using Polly.Extensions.Http;
+using Serilog;
+using Trakx.Utils.Apis;
+
+namespace Trakx.Kaiko.ApiClient;
+
+internal static class HttpClientRegistration
+{
+    internal static IServiceCollection ConfigureHttpClient<TClient,
```
Comment (emanuelgaspar) [inline src/Trakx.Kaiko.ApiClient/Registration/HttpClientRegistration.cs:15]: I took this out of the T4 template generation and it works well. However, Codacy is asking me to add parameters for TClient and TImplementation. 

<https://app.codacy.com/gh/trakx/kaiko-api-client/pullRequest?prid=10893365>

I can't do that.
If there's a different way to write this generic code, please help. If not, can we disable the Codacy alerts for this file?

File: src/Trakx.Kaiko.ApiClient/MarketData/AggregatesClient.cs [hunk 2]
```diff
@@ -0,0 +1,16 @@
+﻿using System.Text;
+
+namespace Trakx.Kaiko.ApiClient;
+
+// this is only here to stop Codacy from flagging issue:
+// 'partial' is gratuitous in this context.
+internal partial class AggregatesClient { }
```
Comment (emanuelgaspar) [inline src/Trakx.Kaiko.ApiClient/MarketData/AggregatesClient.cs:7]: Yes, really. It seems Codacy analyses files one by one, so it reports partial classes as 'gratuitous' incorrectly.

Comment (emanuelgaspar) [inline src/Trakx.Kaiko.ApiClient/MarketData/AggregatesClient.cs:7]: Yes, really. It seems Codacy analyses files one by one, so it reports partial classes as 'gratuitous' incorrectly.

File: src/Trakx.Kaiko.ApiClient/ReferenceData/AssetsClient.cs [hunk 3]
```diff
@@ -0,0 +1,16 @@
+﻿using System.Text;
+
+namespace Trakx.Kaiko.ApiClient;
+
+// this is only here to stop Codacy from flagging issue:
+// 'partial' is gratuitous in this context.
+internal partial class AssetsClient { }
```
Comment (emanuelgaspar) [inline src/Trakx.Kaiko.ApiClient/ReferenceData/AssetsClient.cs:7]: Yes, really. It seems Codacy analyses files one by one, so it reports partial classes as 'gratuitous' incorrectly.

File: src/Trakx.Kaiko.ApiClient/ReferenceData/ExchangesClient.cs [hunk 4]
```diff
@@ -0,0 +1,16 @@
+﻿using System.Text;
+
+namespace Trakx.Kaiko.ApiClient;
+
+// this is only here to stop Codacy from flagging issue:
+// 'partial' is gratuitous in this context.
+internal partial class ExchangesClient { }
```
Comment (emanuelgaspar) [inline src/Trakx.Kaiko.ApiClient/ReferenceData/ExchangesClient.cs:7]: Yes, really. It seems Codacy analyses files one by one, so it reports partial classes as 'gratuitous' incorrectly.

File: src/Trakx.Kaiko.ApiClient.Tests/Integration/GenerateApiClientChecker.cs [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+public class GenerateApiClientChecker
+{
+    public ITestOutputHelper Output { get; }
+
+    public GenerateApiClientChecker(ITestOutputHelper output)
+    {
+        Output = output;
+    }
+
+    [Fact]
```
Comment (monsieurleberre) [inline src/Trakx.Kaiko.ApiClient.Tests/Integration/GenerateApiClientChecker.cs:16]: ❤️ thanks!! can go to Common.Testing soon ? 😉

File: src/Trakx.Kaiko.ApiClient/MarketData/AggregatesClient.cs (hunk 2 above)
Comment (monsieurleberre) [reply src/Trakx.Kaiko.ApiClient/MarketData/AggregatesClient.cs:7]: sorry about that

File: src/Trakx.Kaiko.ApiClient/Registration/HttpClientRegistration.cs (hunk 1 above)
Comment (monsieurleberre) [reply src/Trakx.Kaiko.ApiClient/Registration/HttpClientRegistration.cs:15]: ignored

**Review by monsieurleberre — APPROVED** [review]
Thanks a lot! happy to merge :)

File: src/Trakx.Kaiko.ApiClient.Tests/Integration/GenerateApiClientChecker.cs (hunk 5 above)
Comment (emanuelgaspar) [reply src/Trakx.Kaiko.ApiClient.Tests/Integration/GenerateApiClientChecker.cs:16]: Cool, I'll ask @joaofbvieira to include it as part of his refactor.

**Pull request merged**